### PR TITLE
Fixed bookmarks merge

### PR DESCRIPTION
**Pull request by rokuz**

### CODE REVIEW

File: map/bookmark_manager.cpp [hunk 1]
```diff
@@ -914,21 +914,24 @@ void BookmarkManager::CreateCategories(KMLDataCollection && dataCollection, bool
 {
   ASSERT_THREAD_CHECKER(m_threadChecker, ());
   df::GroupIDSet loadedGroups;
+
+  std::vector<std::pair<df::MarkGroupID, BookmarkCategory *>> categoriesForMerge;
+  categoriesForMerge.reserve(m_categories.size());
+  for (auto const & c : m_categories)
+    categoriesForMerge.emplace_back(std::make_pair(c.first, c.second.get()));
```
Comment (VladiMihaylenko) [inline map/bookmark_manager.cpp]: categoriesForMerge.emplace_back(c.first, c.second.get());

File: map/bookmark_manager.cpp [hunk 2]
```diff
@@ -914,21 +914,24 @@ void BookmarkManager::CreateCategories(KMLDataCollection && dataCollection, bool
...
     df::MarkGroupID groupId;
     BookmarkCategory * group = nullptr;
 
-    auto const it = std::find_if(m_categories.begin(), m_categories.end(),
-      [&data](CategoriesCollection::value_type const & v)
-    {
-      return v.second->GetName() == data->m_name;
-    });
-    bool merge = it != m_categories.end();
+    auto const it = std::find_if(categoriesForMerge.begin(), categoriesForMerge.end(),
```
Comment (VladiMihaylenko) [inline map/bookmark_manager.cpp]: cbegin(), cend()

**Comment by rokuz** [comment]
PTAL @VladiMihaylenko

**Comment by rokuz** [comment]
PTAL @VladiMihaylenko Дудь респекнул)

**Pull request merged**